lib-injection/host_inject.rb

```diff
@@ -0,0 +1,116 @@
+return if ENV["DD_TRACE_SKIP_LIB_INJECTION"] == "true"
+require "rubygems"
+require "rbconfig"
+
+ruby_api_version = RbConfig::CONFIG["ruby_version"]
+
+# Look for pre-installed tracers
+Gem.paths = {
+  "GEM_PATH" => "/opt/datadog/apm/library/ruby/#{ruby_api_version}:/opt/datadog/apm/library/ruby:#{ENV["GEM_PATH"]}"
+}
+
+# Also apply to the environment variable, to guarantee any spawned processes will respected the modified `GEM_PATH`.
+ENV["GEM_PATH"] = Gem.path.join(":")
+
+def debug_log(msg)
+  STDOUT.puts msg if ENV["DD_TRACE_DEBUG"] == "true"
+end
+
+begin
+  require "open3"
+  require "bundler"
+  require "bundler/cli"
+  require "shellwords"
+  require "fileutils"
+
+  failure_prefix = "Datadog lib injection failed:"
+  support_message = "For help solving this issue, please contact Datadog support at https://docs.datadoghq.com/help/."
+
+  unless Bundler::SharedHelpers.in_bundle?
+    debug_log "[ddtrace] Not in bundle... skipping host injection"
+    return
+  end
+
+  unless Bundler::CLI.commands["add"] && Bundler::CLI.commands["add"].options.key?("require")
+    debug_log "[ddtrace] You are currently using Bundler version #{Bundler::VERSION} which is not supported by host injection. Please upgrade >= 2.3, check https://github.com/rubygems/rubygems/blob/master/bundler/CHANGELOG.md#enhancements-31"
+    return
+  end
```

Comment:
I'm not sure the link to the changelog adds a lot -- e.g. what do we want customers to learn from that link?